node/src/actors/chain_manager/mod.rs

```diff
@@ -1152,30 +1154,36 @@ impl ChainManager {
                     // process, which are assumed to be valid.
                     SuperBlockConsensus::SameAsLocal
                 };
+
                 match consensus {
                     SuperBlockConsensus::SameAsLocal => {
-                        // Consensus: persist chain state
-                        log::debug!("Before update: superblock {:?}", act.get_superblock_beacon());
+                        // At this point we need to persist previous_chain_state,
+                        let superblock_beacon =  act.get_superblock_beacon();
+                        log::debug!("Before update: superblock {:?}", superblock_beacon);
+
+                        // Take last beacon from superblock state and use it in current chain_info
                         act.chain_state.chain_info.as_mut().unwrap().highest_superblock_checkpoint =
                             act.chain_state.superblock_state.get_beacon();
+
                         if act.sm_state == StateMachine::Synced {
                             // Persist previous_chain_state with current superblock_state
-                            act.persist_chain_state(ctx);
+                            act.persist_previous_chain_state(ctx);
```

Comment:
Here we call `persist_previous_chain_state` but we don't update `self.previous_state`. This means that consecutive calls to `persist_previous_chain_state` will all persist the same state (the initial one). We must update `self.previous_chain_state` after persisting:

```suggestion
                            act.persist_previous_chain_state(ctx);
                            act.previous_chain_state = act.chain_state.clone();
```

Not sure if we should also add this line to the other places, where `self.previous_chain_state` is updated before.

Reply:
Intredasting. @mariocao what do you think?

Reply:
seems correct...

Reply:
Done!